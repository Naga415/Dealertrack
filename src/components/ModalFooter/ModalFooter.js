import React from "react";
import PropTypes from "prop-types";
import Button from "@coxautokc/fusion-ui-components/lib/Button";
import Row from "@coxautokc/fusion-ui-components/lib/Row";
import Col from "@coxautokc/fusion-ui-components/lib/Col";
import "./ModalFooter.scss";

export default function ModalFooter(props) {
  const { modalName, displayResetBtn, resetData, save, cancel } = props;

  return (
    <div id="ModalFooter">
      <Row id="buttongroup">
        <Col lg={6} md={6} sm={6} className="leftbtn">
          {displayResetBtn && (
            <div
              id={`${modalName}-ModalReset`}
              className="leftbtn"
              onClick={resetData}
            >
              <i className="fa fa-refresh fa-2x" aria-hidden="true" />
            </div>
          )}
        </Col>

        <Col lg={6} md={6} sm={6} className="rightbtns">
          <Button
            className="cancel-btn"
            htmlId={`${modalName}-modalCancel`}
            onClick={cancel}
          >
            Cancel
          </Button>
          <Button
            className="save-btn"
            htmlId={`${modalName}-modalSave`}
            onClick={save}
          >
            Save
          </Button>
        </Col>
      </Row>
    </div>
  );
}

ModalFooter.propTypes = {
  cancel: PropTypes.func.isRequired,
  displayResetBtn: PropTypes.bool,
  modalName: PropTypes.string.isRequired,
  resetData: PropTypes.func,
  save: PropTypes.func.isRequired
};

ModalFooter.default = {
  displayResetBtn: false
};
