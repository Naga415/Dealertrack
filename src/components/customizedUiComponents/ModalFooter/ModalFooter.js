import React from "react";
import PropTypes from "prop-types";
import Button from "@coxautokc/fusion-ui-components/lib/Button";
import Row from "@coxautokc/fusion-ui-components/lib/Row";
import Col from "@coxautokc/fusion-ui-components/lib/Col";
import "./ModalFooter.scss";

export default function ModalFooter(props) {
  const {
    modalName,
    displayResetBtn,
    save,
    cancel,
    isDisabled
  } = props;

  return (
    <div id="ModalFooter">
      <Row id="buttongroup">
        <Col lg={6} md={6} sm={6} className="rightbtns">
          <Button
            className="cancel-btn"
            htmlId={`${modalName}-modalCancel`}
            onClick={cancel}
          >
            Cancel
          </Button>
          <Button
            disabled={isDisabled}
            className="save-btn"
            htmlId={`${modalName}-modalSave`}
            onClick={save}
          >
            Save
          </Button>
        </Col>
      </Row>
    </div>
  );
}

ModalFooter.propTypes = {
  cancel: PropTypes.func.isRequired,
  displayResetBtn: PropTypes.bool,
  isDisabled: PropTypes.bool,
  modalName: PropTypes.string.isRequired,
  save: PropTypes.func.isRequired
};
