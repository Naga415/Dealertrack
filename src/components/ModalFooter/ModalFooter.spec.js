"use strict";
import React from "react";
import sinon from "sinon";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import ModalFooter from "./ModalFooter";
import 'jsdom-global/register';

const props = {
  cancel: sinon.spy(),
  displayResetBtn: true,
  modalName: "modalName",
  resetData: sinon.spy(),
  save: sinon.spy()
};

afterEach(() => {
  props.resetData.reset();
  props.save.reset();
  props.cancel.reset();
});

describe("ModalFooter specs", () => {
  it("renders <ModalFooter />", () => {
    const component = renderer.create(<ModalFooter {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("triggers cancel button when click", () => {
    const component = mount(<ModalFooter {...props} />);
    component.find("#modalName-modalCancel").simulate("click");
    expect(props.cancel.calledOnce).toEqual(true);
  });

  it("triggers save button when click", () => {
    const component = mount(<ModalFooter {...props} />);
    component.find("#modalName-modalSave").simulate("click");
    expect(props.save.calledOnce).toEqual(true);
  });

  it("refresh button not exists when pass false as displayResetBtn", () => {
    props["displayResetBtn"] = false;
    const component = mount(<ModalFooter {...props} />);
    expect(component.find(".fa .fa-refresh").exists()).toEqual(false);
  });
});
