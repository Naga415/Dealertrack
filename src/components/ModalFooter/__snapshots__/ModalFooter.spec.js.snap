// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter specs renders <ModalFooter /> 1`] = `
<div
  id="ModalFooter"
>
  <div
    className="row"
    id="buttongroup"
  >
    <div
      className="leftbtn col-lg-6 col-md-6 col-sm-6"
      id={undefined}
    >
      <div
        className="leftbtn"
        id="modalName-ModalReset"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-refresh fa-2x"
        />
      </div>
    </div>
    <div
      className="rightbtns col-lg-6 col-md-6 col-sm-6"
      id={undefined}
    >
      <button
        className="cancel-btn btn btn-default"
        disabled={false}
        href={undefined}
        id="modalName-modalCancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="save-btn btn btn-default"
        disabled={false}
        href={undefined}
        id="modalName-modalSave"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
